packages/nocodb/src/lib/meta/api/sync/helpers/job.ts

```diff
@@ -2235,9 +2235,9 @@ export default async (
 
         const recordsMap = {};
 
-        for (let i = 0; i < ncTblList.list.length; i++) {
+        for await (const table of ncTblList.list.filter((ncTbl) => tables.find((table) => table.title === ncTbl.title))) {
```

Comment:
Why `for await...of` here, is it required or just ``for await...of` is enough?

Reply:
Yes right also I missed that we only miss that check for link part, so I revised that commit, thank you for pointing out.